packages/core-ethereum/src/indexer/index.ts

```diff
@@ -226,10 +228,21 @@ class Indexer extends EventEmitter {
           continue
         }
       }
+      confirmedEvents.push(event)
+    }
 
-      const eventName = event.event as EventNames
+    // Sort announcements first, so we have a record of address => publickeys
+    // when processing other updates.
+    confirmedEvents.sort((a, b) => {
+      if (a.event === ANNOUNCEMENT) {
+        return b.event === ANNOUNCEMENT ? 0 : -1
+      }
+      return b.event === ANNOUNCEMENT ? 1 : 0
+    })
 
-      if (eventName === 'Announcement') {
+    for (const event of confirmedEvents) {
+      const eventName = event.event as EventNames
+      if (eventName === ANNOUNCEMENT) {
```

Comment:
Why don't use a `switch` block, i.e. `switch (eventName) { .... default: log("unrecognized event") }`

Reply:
unchanged from before. We could change to a switch in future.